### PR TITLE
[pickers] Modify `adapter.isEqual` method to accept `TDate | null` instead of any

### LINKED ISSUE #10974: [pickers] v7: Modify `adapter.isEqual` method to accept `TDate | null` instead of any
Part of #7902

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #10974

### CODE REVIEW

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 1]
```diff
@@ -63,3 +63,48 @@ For example:
 
 The same applies to `slotProps` and `componentsProps`.
 :::
+
+### Adapters
+
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:68]:
```suggestion
### Adapters

The adapter does the glue between the date library you chose and the pickers component.
Standard usage of the library should not include those adapters.
Those breaking changes are listed for users that have advanced usage such as creating a library based on the pickers.

```

Maybe adding a text to say that most of the user don't have to care about this section

Comment (flaviendelangle) [reply docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:68]: Oh sorry, I did a brief introduction in #10971, I'll let you review there if you want to improve the content :+1:

Comment (alexfauquette) [reply docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:68]: No problem, your description is nice in this other PR :+1:

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 2]
```diff
@@ -63,3 +63,48 @@ For example:
 
 The same applies to `slotProps` and `componentsProps`.
 :::
+
+### Adapters
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:67]:
```suggestion
## Adapters
```

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 3]
```diff
@@ -63,3 +63,48 @@ For example:
...
+
+#### Restrict the input format of the `isEqual` method
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:69]:
```suggestion
### Restrict the input format of the `isEqual` method
```

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md (hunk 2 above)
Comment (alexfauquette) [reply docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:67]: From what we discussed in the meeting. But that could be done in one of the PR

**Review by alexfauquette — APPROVED** [review]
The code is much cleaner when assuming we got parsed value or null :+1:

**Pull request merged**